lib/validation/correspondanceAddressManual.ts

```diff
@@ -3,35 +3,53 @@ import correspondanceAddressManualErrorManifest from "../utils/error_manifests/c
+        .isLength({ max: 50 }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressLine2Length.summary),
 
-    body("addressTown").notEmpty().withMessage(correspondanceAddressManualErrorManifest.validation.noCityOrTown.summary).bail()
+    body("addressTown").trim().notEmpty().withMessage(correspondanceAddressManualErrorManifest.validation.noCityOrTown.summary).bail()
         .matches(addressTownFormat).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressTown.summary).bail()
-        .isLength({ min: 1, max: otherAddressDetailsLength }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressTownLength.summary),
-
-    body("addressCounty").matches(addressCountyFormat).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressCounty.summary).bail()
-        .isLength({ max: otherAddressDetailsLength }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressCountyLength.summary),
+        .isLength({ max: 50 }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressTownLength.summary),
 
-    body("addressCountry").matches(addressCountyFormat).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressCountry.summary).bail()
-        .isLength({ max: otherAddressDetailsLength }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressCountryLength.summary),
+    body("addressCounty").custom((value, { req }) => addressCountyChecker(req.body.addressCounty)),
+    body("addressCountry").custom((value, { req }) => addressCountryChecker(req.body.addressCountry)),
 
-    body("addressPostcode").notEmpty().withMessage(correspondanceAddressManualErrorManifest.validation.noPostCode.summary).bail()
+    body("addressPostcode").trim().notEmpty().withMessage(correspondanceAddressManualErrorManifest.validation.noPostCode.summary).bail()
         .matches(addressUKPostcodeFormat).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressPostcode.summary).bail()
-        .isLength({ min: 1, max: otherAddressDetailsLength }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressPostcode.summary)
+        .isLength({ min: 6, max: 50 }).withMessage(correspondanceAddressManualErrorManifest.validation.invalidAddressPostcode.summary)
 ];
+
+export const addressCountyChecker = (addressCounty: string) => {
+    addressCounty = addressCounty.trim();
+    if (addressCounty !== "") {
+        if (!addressCounty.match(addressCountyFormat)) {
+            throw new Error(correspondanceAddressManualErrorManifest.validation.invalidAddressCounty.summary);
+        } else if (addressCounty.length < 5 || addressCounty.length > 50) {
+            throw new Error(correspondanceAddressManualErrorManifest.validation.invalidAddressCountyLength.summary);
+        }
+    }
+    return true;
+};
+
+export const addressCountryChecker = (addressCountry: string) => {
```

Comment:
addressCountryChecker and addressCountyChecker functions do the same thing, could these be combined to a single function

Reply:
comments addressed